common/metrics/simplereporter.go

```diff
@@ -36,6 +40,11 @@ type (
 	}
 )
 
+type HandlerFn func(metricName string, baseTags, tags map[string]string, value int64)
+
+var handlers = make(map[string]map[string]HandlerFn) // Key1 - metricName; Key2 - "filterTag:filterVal"
+var handlerMutex sync.RWMutex
+
```

Comment:
Shouldn't `handlers` and `handlerMutex` be part of `SimpleReporter` struct? If someone creates two `SimpleReporter` instances the handlers will step on each other.